flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -310,6 +310,12 @@ class TaskManager(
 
     case FatalError(message, cause) =>
       killTaskManagerFatal(message, cause)
+
+    case RequestTaskManagerLog(requestType : LogTypeRequest) =>
+      handleRequestTaskManagerLog(sender(), requestType, currentJobManager.get)
+
+    case IsBlobServiceDefined() =>
+      sender() ! (blobService.isDefined && currentJobManager.get.path.address.host.isDefined)
```

Comment:
if the `blobService` is defined, then it should have a valid `serverAddress` where the `BlobServer` is running.